### DISCUSSION TITLE
What is the correct way to load the same Soundfont into multiple synths?

### DISCUSSION THREAD

**Discussion by JBetz** [FluidSynth/fluidsynth#1210] [Q&A] [answered]
If you want multiple synths to use the same Soundfont, what is the difference between:

1. Calling sfload on each synth
2. Calling sfload on the first synth, then using add_sfont with the sfont pointer on all subsequent synths

Second seems like it would be more efficient, but might have unintended side effects if the Soundfont state is shared between synths.

**Comment by derselbst** [comment, marked as answer]
Fluidsynth uses a sample cache for all loaded soundfonts. Therefore, reusing the same font between synths is cheap. 1. is your only option.

Option 2. results in undefined behavior, because the sfont is already being owned by another synth.

**Reply by JBetz** [reply]
Perfect, thanks!
